app/workers/irker_worker.rb

```diff
@@ -0,0 +1,127 @@
+      return false
+    end
+
+    repo_name     = push_data["repository"]["name"];
+    committer     = push_data["user_name"]
+    branch        = push_data["ref"].gsub(%r"refs/[^/]*/", "")
+    commits_count = push_data["total_commits_count"]
+
+    if colors
+      repo_name     = "\x0304#{repo_name}\x0f"
+      branch        = "\x0305#{branch}\x0f"
+      commits_count = "\x02#{commits_count}\x0f"
+    end
+
+
+    # Firsts messages are for branch creation/deletion
+    if push_data["before"] =~ /^000000/
+      newbranch = "#{Gitlab.config.gitlab.url}/#{path_with_namespace}/branches"
+      if colors
+        newbranch = "\x0302\x1f#{newbranch_url}\x0f"
+      end
+
+      privmsg  = "[#{repo_name}] #{committer} has created a new branch "
+      privmsg += "#{branch}: #{newbranch}"
+      sendtoirker privmsg
+    elsif push_data["after"] =~ /^000000/
+      privmsg = "[#{repo_name}] #{committer} has deleted the branch #{branch}"
+      sendtoirker privmsg
+    end
+
+    # Next message is for number of commit pushed, if any
+    if push_data["total_commits_count"] == 0
+      return true
+    end
+
+    if push_data["before"] =~ /^000000/
+      # Tweak on push_data["before"] in order to have a nice compare URL
+      commit_id = push_data["commits"][0]["id"]
+      commit = Gitlab::Git::Commit.find(project.repository, commit_id)
+      commit = Commit.new(commit)
+      parents = commit.parents
+      push_data["before"] = parents[0].id unless parents.empty?
+    end
+
+    sha1         = Commit::truncate_sha(push_data["before"])
+    sha2         = Commit::truncate_sha(push_data["after"])
+    compare_url  = "#{Gitlab.config.gitlab.url}/#{path_with_namespace}/compare"
+    compare_url += "/#{sha1}...#{sha2}"
+    compare_url  = "\x0302\x1f#{compare_url}\x0f" if colors
+
+    new_commits  = "new commit"
+    new_commits += 's' if push_data["total_commits_count"] > 1
+
+    privmsg      = "[#{repo_name}] #{committer} pushed #{commits_count} "
+    privmsg     += "#{new_commits} to #{branch}: #{compare_url}"
+    sendtoirker privmsg
+
+    # Finally, one message per commit, limited by 3 messages (same limit as the
+    # github irc hook)
+    commits = push_data["commits"].first(3)
+    commits.each do |hook_attrs|
+      commit = Gitlab::Git::Commit.find(project.repository, hook_attrs["id"])
+      commit = Commit.new(commit)
+      sha    = Commit::truncate_sha(hook_attrs["id"])
+      author = hook_attrs["author"]["name"]
+      files  = "#{commit.diffs.count} file"
+      files += 's' if commit.diffs.count > 1
+      title  = commit.title
+
+      if colors
+        sha = "\x0314#{sha}\x0f"
+        files = "\x0312#{files}\x0f"
+      end
+
+      privmsg = "#{repo_name}/#{branch} #{sha} #{author} (#{files}): #{title}"
+      sendtoirker privmsg
+    end
+
+    @socket.close
+
+    return true
+  ensure
+    GC.start
+  end
+
+  def sendtoirker(privmsg)
+    @socket.puts JSON.dump({
+      to: @channels,
+      privmsg: privmsg
+    })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.